packages/core/src/render/RenderSystem.ts

```diff
@@ -0,0 +1,126 @@
+import { Renderer } from '../Renderer';
+import { RenderTexture } from '../renderTexture/RenderTexture';
+
+/**
+ * system that provides a render function that focussing on rendering Pixi Scene Graph objects
+ * to either the main view or to a renderTexture.  Used for Canvas `WebGL` contexts
+ *
+ * @memberof PIXI
+ */
+export class RendererSystem implements ISystem
+{
+    renderer: Renderer;
+
+    /**
+     * Flag if we are rendering to the screen vs renderTexture
+     *
+     * @readonly
+     * @default true
+     */
+    renderingToScreen: boolean;
+
+    /**
+     * the last object rendered by the renderer. Useful for other plugins like interaction managers
+     *
+     * @readonly
+     */
+    lastObjectRendered: IRenderableObject;
+
+    // renderers scene graph!
+    constructor(renderer: Renderer)
+    {
+        this.renderer = renderer;
+    }
+
+    /**
+     * Renders the object to its WebGL view.
+     *
+     * @param displayObject - The object to be rendered.
+     * @param options - the options to be passed to the renderer
+     */
+    render(displayObject: IRenderableObject, options?: IRendererRenderOptions): void
+    {
+        const renderer = this.renderer;
+
+        let renderTexture: RenderTexture;
+        let clear: boolean;
+        let transform: Matrix;
+        let skipUpdateTransform: boolean;
+
+        if (options)
+        {
+            renderTexture = options.renderTexture;
+            clear = options.clear;
+            transform = options.transform;
+            skipUpdateTransform = options.skipUpdateTransform;
+        }
+
+        // can be handy to know!
+        this.renderingToScreen = !renderTexture;
+
+        renderer.runners.prerender.emit();
+        renderer.emit('prerender');
+
+        // apply a transform at a GPU level
+        renderer.projection.transform = transform;
+
+        // no point rendering if our context has been blown up!
+        if (renderer.context.isLost)
+        {
+            return;
+        }
+
+        if (!renderTexture)
+        {
+            this.lastObjectRendered = displayObject;
+        }
+
+        if (!skipUpdateTransform)
+        {
+            // update the scene graph
+            const cacheParent = displayObject.enableTempParent();
+
+            displayObject.updateTransform();
+            displayObject.disableTempParent(cacheParent);
+            // displayObject.hitArea = //TODO add a temp hit area
+        }
+
+        renderer.renderTexture.bind(renderTexture);
+        renderer.batch.currentRenderer.start();
+
+        if (clear !== undefined ? clear : renderer.background.clearBeforeRender)
+        {
+            renderer.renderTexture.clear();
+        }
+
+        displayObject.render(renderer);
+
+        // apply transform..
+        renderer.batch.currentRenderer.flush();
+
+        if (renderTexture)
+        {
+            // @dev7355608 please confirm this is ok to be here?
+            renderer.framebuffer.blit();
```

Comment:
The blit should be optional, because there are cases where you might want to blit to some other texture instead (see #6844). If the default is set to true, we need to remove the blit in `Renderer#generateTexture` and cache-as-bitmap to avoid an unnecessary second blit. The documentation of the `multisample` property also needs to be updated.

Reply:
thanks @dev7355608 ! I will update

Reply:
done! mind taking a second look pls @dev7355608 ? Cheers!